pkg/commatrix-creator/commatrix_test.go

```diff
@@ -0,0 +1,278 @@
+						"app":                        "test-app",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "test-container",
+							Image: "test-image:latest",
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 80,
+								},
+							},
+						},
+					},
+				},
+			}
+
+			testService := &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-service",
+					Namespace: "test-ns",
+					Labels: map[string]string{
+						"kubernetes.io/service-name": "test-service",
+					},
+				},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{
+						"app": "test-app",
+					},
+					Ports: []corev1.ServicePort{
+						{
+							Port:     80,
+							Protocol: corev1.ProtocolTCP,
+						},
+					},
+					Type: corev1.ServiceTypeNodePort,
+				},
+			}
+
+			testEndpointSlice := &discoveryv1.EndpointSlice{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-service-endpoints",
+					Namespace: "test-ns",
+					Labels: map[string]string{
+						"kubernetes.io/service-name": "test-service",
+					},
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							Kind: "Service",
+							Name: "test-service",
+						},
+					},
+				},
+				AddressType: discoveryv1.AddressTypeIPv4,
+				Endpoints: []discoveryv1.Endpoint{
+					{
+						NodeName:  &testNode.Name,
+						Addresses: []string{"192.168.1.1", "192.168.1.2"},
+					},
+				},
+				Ports: []discoveryv1.EndpointPort{
+					{
+						Name:     nil,
+						Port:     func(i int32) *int32 { return &i }(80),
+						Protocol: func(p corev1.Protocol) *corev1.Protocol { return &p }(corev1.ProtocolTCP),
+					},
+				},
+			}
+
+			fakeClient := fake.NewClientBuilder().WithScheme(sch).WithObjects(testNode, testPod, testService, testEndpointSlice).Build()
+			fakeClientset := fakek.NewSimpleClientset()
+
+			clientset := &client.ClientSet{
+				Client:          fakeClient,
+				CoreV1Interface: fakeClientset.CoreV1(),
+			}
+
+			testEpsComDetails = []types.ComDetails{
+				{
+					Direction: "Ingress",
+					Protocol:  "TCP",
+					Port:      80,
+					Namespace: "test-ns",
+					Service:   "test-service",
+					Pod:       "test-pod",
+					Container: "test-container",
+					NodeRole:  "master",
+					Optional:  false,
+				},
+			}
+
+			endpointSlices, err = endpointslices.New(clientset)
+			o.Expect(err).ToNot(o.HaveOccurred())
+		})
+
+		g.It("Should successfully create an endpoint matrix with custom entries", func() {
+			commatrixCreator, err := New(endpointSlices, "../../samples/custom-entries/example-custom-entries.csv", types.FormatCSV, types.Cloud, types.SNO)
+			o.Expect(err).ToNot(o.HaveOccurred())
+			commatrix, err := commatrixCreator.CreateEndpointMatrix()
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			wantedComDetails := append(append(append(testEpsComDetails, types.CloudStaticEntriesMaster...), types.GeneralStaticEntriesMaster...), exampleComDetailsList...)
```

Comment:
add exampleComDetailsList on a separate line to clarify maybe also a separate By()